### PR TITLE
Add exit status code on validation failure

### PR DESCRIPTION
**Pull request by awright415**
For CI purposes, it'd be useful to ensure that failed validations return an exit code of 1.

### CODE REVIEW

**Comment by thebinarypenguin** [comment]
I agree about the exit code, but I would prefer to have the error counter in the main `raml-cop.js` instead of `reporter.js`.

File: test/stdio.test.js [hunk 1]
```diff
@@ -38,29 +38,45 @@ describe('STDIO', function() {
   describe('When called with STDIN', function() {
 
     describe('Valid STDIN', function() {
-      var validStream = fs.createReadStream(validFile);
+      var validStream = function() { return fs.createReadStream(validFile) };
```
Comment (thebinarypenguin) [inline test/stdio.test.js]: I don't understand why you made this a function

File: test/stdio.test.js [hunk 2]
```diff
@@ -38,29 +38,45 @@ describe('STDIO', function() {
...
+          if (err) { done(err); }
+          chai.expect(code).to.eq(0);
+          done();
+        });
+      });
     });
 
     describe('Invalid STDIN', function() {
-      var invalidStream = fs.createReadStream(invalidFile);
+      var invalidStream = function() { return fs.createReadStream(invalidFile); }
```
Comment (thebinarypenguin) [inline test/stdio.test.js]: I don't understand why you made this a function

File: test/stdio.test.js (hunk 1 above)
Comment (awright415) [inline test/stdio.test.js]: Was having trouble with the same stream being used for multiple tests

**Comment by awright415** [comment]
Was trying to work with what was already there, but perhaps having an exit code variable that gets flipped to 1 in the error callback instead of using reporter.js?

**Comment by thebinarypenguin** [comment]
I moved the error counter into the main `raml-cop.js` file and updated the tests. Does it look good to you?

**Pull request closed without merging**